src/cup1d/likelihood/emcee_sampler.py

```diff
@@ -77,876 +120,1052 @@ def __init__(self,like=None,
         # Figure out what extra information will be provided as blobs
         self.blobs_dtype = self.like.theory.get_blobs_dtype()
 
-
     def set_truth(self):
-        """ Set up dictionary with true values of cosmological
-        likelihood parameters for plotting purposes """
+        """Set up dictionary with true values of cosmological
+        likelihood parameters for plotting purposes"""
 
         # likelihood contains true parameters, but not in latex names
-        like_truth=self.like.truth
+        like_truth = self.like.truth
 
         # when running on data, we do not know the truth
         if like_truth is None:
-            self.truth=None
+            self.truth = None
             return
 
         # store truth for all parameters, with LaTeX keywords
-        self.truth={}
+        self.truth = {}
         for param in cosmo_params:
-            param_string=param_dict[param]
-            self.truth[param_string]=like_truth[param]
+            param_string = param_dict[param]
+            self.truth[param_string] = like_truth[param]
 
-        return
+        if (self.like.data.sim_label[4:].isdigit() == False) & (
```

Comment:
Return truth value for IGM parameters for fiducial simulations

Reply:
What would happen when analysing other test simulations, like neutrinos or growth? We do not know the true IGM parameters for these either

Reply:
We do now, I updated the code after merging the PR to LaCE.

Reply:
What do you mean? How do we know the true IGM parameters for any simulation that is not the fiducial?

Reply:
Sorry, my previous comment was not clear. When creating a mock P1D measurement from a mock, we now read the true IGM history (the one of the simulation we use as input, if available) and a fiducial one that we assume for fitting. I agree that we do not know the value of the true IGM parameters, but we can plot the true IGM history, the fiducial one, and the best-fitting one. It is a good way of knowing whether the best-fitting IGM history makes sense; looking at the best-fitting parameters is not very informative. I attach an example. It is a very noisy test just for checking the pipeline is working. I am attempting to fit mpg_5 using as fiducial the IGM history of mpg_reio.
[IGM_histories_mean.pdf](https://github.com/igmhub/cup1d/files/13536682/IGM_histories_mean.pdf)